app/recordtransfer/views/profile.py

```diff
@@ -13,96 +13,140 @@
+    def get_error_message(self) -> str:
+        """Return the error message for the specific form type."""
+        raise NotImplementedError("Subclasses must implement get_error_message")
 
     def form_valid(self, form: BaseModelForm) -> HttpResponse:
-        """Handle valid form submission."""
-        response = super().form_valid(form)
-        message = self.success_message
-        if form.cleaned_data.get("new_password"):
-            update_session_auth_hash(self.request, form.instance)
-            message = self.password_change_success_message
-
-            context = {
-                "subject": gettext("Password updated"),
-                "changed_item": gettext("password"),
-                "changed_status": gettext("updated"),
-            }
-            send_user_account_updated.delay(self.get_object(), context)
-
-        messages.success(self.request, message)
-        return response
+        """Handle valid form submission, return updated form and trigger success event in
+        response.
+        """
+        try:
+            super().form_valid(form)
+            if form.cleaned_data.get("new_password"):
+                update_session_auth_hash(self.request, form.instance)
+            context = self.get_context_data(form=form)
+            response = self.render_to_response(context)
+            return trigger_client_event(
+                response, "showSuccess", {"value": self.get_success_message()}
+            )
+        except Exception:
+            LOGGER.exception("Failed to update user information for user %s", self.request.user.id)
+            response = HttpResponse(status=500)
+            return trigger_client_event(
+                response, "showError", {"value": gettext(self.get_error_message())}
+            )
 
     def form_invalid(self, form: BaseModelForm) -> HttpResponse:
         """Handle invalid form submission."""
-        messages.error(
-            self.request,
-            self.error_message,
+        super().form_invalid(form)
+        context = self.get_context_data(form=form)
+        response = self.render_to_response(context)
+        return trigger_client_event(
+            response, "showError", {"value": gettext("Please correct the errors below.")}
         )
-        profile_form = cast(UserProfileForm, form)
-        profile_form.reset_form()
-        return super().form_invalid(profile_form)
+
+
+class AccountInfoUpdateView(BaseUserProfileUpdateView):
+    """View to update user account information such as name and notification preferences."""
+
+    form_class = UserProfileForm
+    template_name = "includes/account_info_form.html"
+
+    def get_success_message(self) -> str:
+        """Return a success message for account information updates."""
+        return "Account details updated."
+
+    def get_error_message(self) -> str:
+        """Return an error message for account information updates."""
+        return "Failed to update account information."
```

Comment:
From my above suggestion:

```suggestion
class AccountInfoUpdateView(BaseUserProfileUpdateView):
    """View to update user account information such as name and notification preferences."""

    form_class = UserProfileForm
    template_name = "includes/account_info_form.html"
    update_success_message = gettext("Account details updated.")
    update_error_message = gettext("Failed to update account information.")
```